### PR TITLE
manual build via rollup

### PR DESCRIPTION
**Pull request by phryneas**
Putting this up here to see if CSB & co work. Still got some work to do.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -62,15 +62,38 @@
...
-      "path": "dist/rtk-query.cjs.production.min.js",
-      "limit": "25 KB"
+      "name": "ESM full",
+      "path": "dist/esm/index.js",
+      "limit": "15 KB"
+    },
+    {
+      "name": "createApi + setupListeners",
+      "path": "dist/esm/index.js",
+      "import": "{ createApi, setupListeners }"
```
Comment (markerikson) [inline package.json:72]: Oh wow, I had no idea this was a thing!  That's really cool!

Comment (phryneas) [reply package.json:72]: I will, as soon as I recreate the changes I did here over there ;)

File: rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      typescript(defaultTsConfig),
+      babel({
+        exclude: 'node_modules/**',
+        extensions: ['.js', '.ts'],
+        babelHelpers: 'runtime',
+        presets: [
+          [
+            '@babel/preset-env',
+            {
+              targets: { node: true, browsers: ['defaults', 'not IE 11', 'maintained node versions'] },
```
Comment (phryneas) [inline rollup.config.js:50]: So this now has two builds: an ESM build that does not bundle up the transpiled files, so just a file-for-file transpilation. This should help modern bundlers with tree-shaking quite a lot.
The target I chose for that is "all non-dead browsers, last 2 versions, no IE, maintained node versions".

The CJS build includes IE as well and builds only one file.

Do you think that's sensible @markerikson ?

Comment (phryneas) [reply rollup.config.js:50]: For reference: including IE there bumps the bundle sizes for the ESM module up to

```
  ESM full
  Size limit: 15 KB
  Size:       13.8 KB with all dependencies, minified and gzipped
  
  createApi + setupListeners
  Size: 12.55 KB with all dependencies, minified and gzipped
  
  fetchBaseQuery
  Size: 3.23 KB with all dependencies, minified and gzipped
  
  retry
  Size: 2.79 KB with all dependencies, minified and gzipped
  
  ApiProvider
  Size: 387 B with all dependencies, minified and gzipped
```

Comment (markerikson) [reply rollup.config.js:50]: Sure, fine by me.

File: rollup.config.js [hunk 3]
```diff
@@ -74,6 +75,11 @@ const configs = [
           ? defaultTsConfig
           : { ...defaultTsConfig, declarationDir: 'dist/ts', declaration: true, declarationMap: true }
       ),
+      replace({
+        values: {
+          'process.env.NODE_ENV': JSON.stringify(minfied ? 'production' : 'development'),
```
Comment (markerikson) [inline rollup.config.js:80]: typo here? `minfied`

Comment (phryneas) [reply rollup.config.js:80]: yah, but just a variable name, will work either way :rofl:

**Comment by phryneas** [comment]
I guess letting this sit around any longer won't do any good -merging it in :)

**Pull request merged**